sale_order_project/wizard/sale_order_createproject.py

```diff
@@ -0,0 +1,36 @@
+from odoo import api, fields, models, _
+
+
+class SaleOrderCreateProject(models.TransientModel):
+    """ wizard to create a Project from a Sale Order """
+    _name = "sale.order.createproject"
+
+    @api.model
+    def default_get(self, fields):
+        result = super(SaleOrderCreateProject, self).default_get(fields)
+        sale_order_id = self.env.context.get('active_id')
+        if sale_order_id:
+            result['sale_order_id'] = sale_order_id
+        return result
+
+    sale_order_id = fields.Many2one('sale.order', string='Order', domain=[('type', '=', 'order')])
+    related_project_id = fields.Many2one('project.project', string='Project')
+
+    @api.multi
+    def action_create_project_task(self):
+        self.ensure_one()
+        # get the lead to transform
+        order = self.sale_order_id
+        project = self.related_project_id
+        """
+        # if related_project_id is empty
```

Comment:
Here is where you can place:
```if project and not order.related_project_id:
    # Within this if the user picked a project and the order does not have a related one
else:
    # Within this else are the other cases, here you can create it

Reply:
I did, please, review my other comments.

Reply:
With that conditional, it still tries to create a new project instead just update the order.

Reply:
```bash
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 931, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 923, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 687, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 672, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3350, in create
    vals = self._add_missing_default_values(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1602, in _add_missing_default_values
    for name, value in defaults.items():
AttributeError: 'NoneType' object has no attribute 'items'
```

Reply:
One, moment, i made a mistake in the `wizard` button.

Reply:
No information in web, i cannot understand it.

Reply:
Check `sale` module wizards. You have one called `sale.make.invoice.advance`. Try to comprehend it completely and then compare it to yours. I'm not able to clone your fork and test it right now, sorry.